File: customers-micronaut/src/main/java/org/mvillarreal/api/dtos/AddressDTO.java
package org.mvillarreal.api.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AddressDTO {
    @NotNull
    @NotEmpty
    private String department;

    @NotNull
    @NotEmpty
    private String province;

    @NotNull
    @NotEmpty
    private String district;

    @NotNull
    @NotEmpty
    private String address;

    @NotNull
    private Double latitude;

    @NotNull
    private Double longitude;
}

File: customers-nest/src/customers/dto/address.dto.ts
import { IsNotEmpty, IsLongitude, IsLatitude } from "class-validator";

export class AddressDTO {
  @IsNotEmpty()
  department: string;
  
  @IsNotEmpty()
  province: string;
  
  @IsNotEmpty()
  district: string;
  
  @IsNotEmpty()
  address: string;
  
  @IsNotEmpty()
  @IsLatitude()
  latitude: number;

  @IsNotEmpty()
  @IsLongitude()
  longitude: number;
}
